### PR TITLE
bpo-36292:  Mark unreachable code as such

### PR DESCRIPTION
**Pull request by stratakis**
https://bugs.python.org/issue36292

### CODE REVIEW

File: Objects/longobject.c [hunk 1]
```diff
@@ -4638,6 +4638,8 @@ long_bitwise(PyLongObject *a,
         break;
     default:
         PyErr_BadArgument();
```
Comment (vstinner) [inline Objects/longobject.c]: This is deadcode: long_bitwise() is a private function only called with '&', '^' or '|'. So I would prefer to use "Py_UNREACHABLE();" here.

File: Objects/longobject.c [hunk 2]
```diff
@@ -4666,6 +4668,9 @@ long_bitwise(PyLongObject *a,
...
```
Comment (vstinner) [inline Objects/longobject.c]: Ditto: use "Py_UNREACHABLE();" instead. No need to both with references. It's deadcode.

**Review by vstinner — CHANGES_REQUESTED** [review]
You are modiying deadcode, it doesn't deserve a NEWS entry :-)

**Comment by terryjreedy** [comment]
The news entry was deleted and `Py_UNREACHABLE()` used instead.  I don't know if our convention is to add the somewhat redundant `/* Unreachable code */`.  Either way, this seems ready to merge.

File: Objects/longobject.c [hunk 3]
```diff
@@ -4637,8 +4637,8 @@ long_bitwise(PyLongObject *a,
         size_z = negb ? size_b : size_a;
         break;
     default:
-        PyErr_BadArgument();
-        return NULL;
+        /* Unreachable code */
```
Comment (vstinner) [inline Objects/longobject.c]: I concur with @terryjreedy: the comment (here and above) is redudant, please remove it.

**Review by vstinner — CHANGES_REQUESTED** [review]

**Review by vstinner — APPROVED** [review]

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**